### PR TITLE
Add German translation

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR includes 2 things:

1) Adds translation templates for all mods (mostly generated by `findtext.lua`)
2) German translation

### CODE REVIEW

File: mods/carts/locale/carts.de.tr [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# textdomain: carts
+Cart=Lore
+(Sneak+Click to pick up)=(Schleichen u. klicken zum Aufheben)
```
Comment (sfan5) [inline mods/carts/locale/carts.de.tr]:
```suggestion
(Sneak+Click to pick up)=(Schleichen u. Klicken zum Aufheben)
```

File: mods/default/locale/default.de.tr [hunk 2]
```diff
@@ -0,0 +1,205 @@
+# textdomain: default
+Locked Chest=Abgeschlossene Truhe
+Locked Chest (owned by @1)=Abgeschlossene Truhe (Eigentümer: @1)
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]: I think it would be nice if this said `(Eigentum von @1)`

File: mods/default/locale/default.de.tr [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+You do not own this chest.=Ihnen gehört diese Truhe nicht
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]:
```suggestion
You do not own this chest.=Ihnen gehört diese Truhe nicht.
```

File: mods/default/locale/default.de.tr [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+a locked chest=eine abgeschlossene Truhe
+Chest=Truhe
+Stick=Stock
+Paper=Papier
+"@1" by @2=„@1“ von @2
+Book=Buch
+Book with Text=Buch mit Text
+Skeleton Key=Generalschlüssel
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]: should match the original in naming, not just "intent"
so e.g. `Skelettschlüssel`

File: mods/default/locale/default.de.tr [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+Tin Ingot=Zinnbarren
+Bronze Ingot=Bronzebarren
+Gold Ingot=Goldbarren
+Mese Crystal Fragment=Mesekristallfragment
+Clay Brick=Tonziegel
+Obsidian Shard=Obsidianscherbe
+Flint=Feuerstein
+Blueberries=Blaubeeren
+Furnace is empty=Ofen ist leer
+100% (output full)=100% (Ausgabe ist voll)
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]: the english text is not a full sentence either
```suggestion
100% (output full)=100% (Ausgabe voll)
```

File: mods/default/locale/default.de.tr [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+Tin Block=Zinnblock
+Bronze Block=Bronzeblock
+Mese Ore=Meseerz
+Mese Block=Meseblock
+Gold Ore=Golderz
+Gold Block=Goldblock
+Diamond Ore=Diamanterz
+Diamond Block=Diamantblock
+Cactus=Kaktus
+Large Cactus Seedling=Großer Kaktussämling
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]: why `-sämling` and not `-setzling` here?

File: mods/default/locale/default.de.tr [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+Apple Wood Fence Rail=Apfelholzzaungeländer
+Acacia Wood Fence Rail=Akazienholzzaungeländer
+Jungle Wood Fence Rail=Dschungelholzzaungeländer
+Pine Wood Fence Rail=Kiefernholzzaungeländer
+Aspen Wood Fence Rail=Espenholzzaungeländer
+Glass=Glas
+Obsidian Glass=Obsidianglas
+Brick Block=Ziegelblock
+Mese Lamp=Meselampe
+Mese Post Light=Mesestandlampe
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]: `Meselampenpfahl` ?

File: mods/default/locale/default.de.tr [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+Diamond Axe=Diamantaxt
+Wooden Sword=Holzschwert
+Stone Sword=Steinschwert
+Bronze Sword=Bronzeschwert
+Steel Sword=Stahlschwert
+Mese Sword=Meseschwert
+Diamond Sword=Diamantschwert
+Key=Schlüssel
+Torch=Fackel
+@1 will intersect protection on growth.=@1 wird mit Schutzzone beim Wachstum überlappen.
```
Comment (sfan5) [inline mods/default/locale/default.de.tr]:
```suggestion
@1 will intersect protection on growth.=@1 wird bei Wachstum mit geschützter Zone überlappen.
```

File: mods/doors/locale/doors.de.tr [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# textdomain: doors
+Hidden Door Segment=Verborgenes Türsegment
+Owned by @1=Eigentümer: @1
+You do not own this locked door.=Diese abgeschlossene Tür gehört Ihnen nicht.
+a locked door=eine abgeschlossene Tür
+Wooden Door=Holztür
+Steel Door=Stahltür
+Glass Door=Glastür
+Obsidian Glass Door=Obsidianglastür
+Owned by @1=Eigentümer: @1
```
Comment (sfan5) [inline mods/doors/locale/doors.de.tr]: same here (`Eigentum von @1`)

File: mods/game_commands/locale/game_commands.de.tr [hunk 10]
```diff
@@ -0,0 +1,4 @@
+# textdomain: game_commands
+Kill yourself to respawn=Selbstmord begehen, um zu respawnen
```
Comment (sfan5) [inline mods/game_commands/locale/game_commands.de.tr]:
```suggestion
Kill yourself to respawn=Selbstmord begehen, um zu Respawnen
```

File: mods/game_commands/locale/game_commands.de.tr [hunk 11]
```diff
@@ -0,0 +1,4 @@
...
+No static_spawnpoint defined=static_spawnpoint ist nicht definiert
```
Comment (sfan5) [inline mods/game_commands/locale/game_commands.de.tr]: I'd prefer either `... nicht definiert worden` or `Kein static_spawnpoint definiert`

File: mods/sethome/locale/sethome.de.tr [hunk 12]
```diff
@@ -0,0 +1,8 @@
+# textdomain: sethome
+Can use /sethome and /home=Kann /sethome und /home benutzen
+Teleport you to your home point=Teleportieren Sie sich zu Ihrem Zuhause-Punkt
+Teleported to home!=Nach Hause teleportiert
```
Comment (sfan5) [inline mods/sethome/locale/sethome.de.tr]:
```suggestion
Teleported to home!=Nach Hause teleportiert!
```

File: mods/stairs/locale/stairs.de.tr [hunk 13]
```diff
@@ -0,0 +1,147 @@
+# textdomain: stairs
+Inner =Innere 
+Outer =Äußere
```
Comment (sfan5) [inline mods/stairs/locale/stairs.de.tr]: these two no longer exist

**Comment by sfan5** [comment]
Good job on the translation and thanks for contributing it.

Now for some general choices:
> Item translated as `Gegenstand`

Personally I would have left this as `Item`.

> adressing the player using `Sie`

I definitely prefer `Du` here, I agree with [this forum post](https://forum.minetest.net/viewtopic.php?f=21&t=23257#p356701).

**Comment by Wuzzy2** [comment]
> (tons of comments)

Most of them are fixed. I probably overlooked something due to the way how GitHub displays things … Will squash everything if people are happy.

> why -sämling and not -setzling here?
> Just wondering.

Why Seedling and not Sapling here?
Just wondering.

> should match the original in naming, not just "intent"
> so e.g. Skelettschlüssel

I just checked again and indeed “Skelettschlüssel” is indeed real word in German (even if somewhat obscure). I thought that it was a false friend. OK.

> Now for some general choices:
> 
> > Item translated as Gegenstand
> 
> Personally I would have left this as Item.
> 
> > adressing the player using Sie
> 
> I definitely prefer Du here, I agree with this forum post.

We have a translation guideline for German, which has been in effect for years. https://dev.minetest.net/Translating/de

I will not change on a single core dev opinion. I wait for more core dev opinions.

Also note that Minetest is currently on “Sie”. Forcing “du” in MTG would introduce an inconsistency, which is especially ugly since MTG is default. A change would require a change in MT, too.

**Comment by sfan5** [comment]
> Will squash everything if people are happy.

Please do.
> > why -sämling and not -setzling here?
> > Just wondering.
> 
> Why Seedling and not Sapling here?
> Just wondering.

The English text makes the same pointless distinction, didn't notice that.

> We have a translation guideline for German, which has been in effect for years. https://dev.minetest.net/Translating/de

Oh that's good to know, leaving it is fine then.

**Comment by Wuzzy2** [comment]
I wait for https://github.com/minetest/minetest_game/pull/2487 to be merged so I can update.

**Comment by Wuzzy2** [comment]
OK, German translations updated. Ready for re-review (and possibly merge).

File: mods/beds/locale/beds.de.tr [hunk 14]
```diff
@@ -0,0 +1,8 @@
+# textdomain: beds
+Fancy Bed=Schickes Bett
+Simple Bed=Schlichtes Bett
+Leave Bed=Bett verlassen
+Good morning.=Guten Morgen.
+@1 of @2 players are in bed=@1 von @2 Spielern sind im Bett
+Force night skip=Nachtüberspringung erzwingen
```
Comment (sfan5) [inline mods/beds/locale/beds.de.tr]: sounds unnecessarily complicated
```suggestion
Force night skip=Überspringen der Nacht erzwingen
```

**Comment by paramat** [comment]
Explanation: 'Sapling' isn't suitable for a cactus, only trees. 'Seedling' is used often for young cacti.

**Comment by h4ml3t** [comment]
Note: this is lacking some templates and translations.  
The affected mods are:  Stairs, TNT, Vessels, Walls, Wool, XPanes

**Comment by Wuzzy2** [comment]
I can not confirm h4ml3t's claim. None of these mods seem to be affected to me. They all have translations + templates.

Note the listed mods are the 6 last mods if you list all mods in alphabetical order. Weird …

**Comment by h4ml3t** [comment]
My bad: NoScript was hiding them.
Sorry :)

File: mods/default/locale/default.de.tr [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+Dry Dirt=Trockene Erde
+Dry Dirt with Dry Grass=Trockene Erde mit trockenem Gras
+Permafrost=Permafrost
+Permafrost with Stones=Permafrost mit Steinen
+Permafrost with Moss=Permafrost mit Moos
+Sand=Sand
+Desert Sand=Wüstensand
+Silver Sand=Silbersand
+Gravel=Kies
+Clay=Ton
```
Comment (SmallJoker) [inline mods/default/locale/default.de.tr]: Lehm/Tonerde? To me those would be clearer, with less risk of misinterpretations.

**Comment by Wuzzy2** [comment]
@SmallJoker: Do you insist on changing the Clay translation?
Also, is this your final complaint or did I overlook something?

**Pull request merged**